spike/eventloop/datastore.go

```diff
@@ -0,0 +1,41 @@
+package eventloop
+
+import (
+	"container/heap"
+	"time"
+)
+
+// HeightTs combines a timestamp and a height.
+type HeightTs struct {
+	TimeStamp time.Time
+	Height    int64
+}
+
+// MinHeap implements a min-heap of HeightTs based on TimeStamp.
+type MinHeap []HeightTs
```

Comment:
I think this might be over-complicating things a bit. For storage, you just need Height -> State, which can be a simple map

I understand you use this for the rotation stuff, but it's not strictly necessary here, so I would rather we either just drop the data rotation entirely, or you can simply add rotation on a X number of block basis rather than time. If you do, if would still prefer to keep that code out of the eventloop file, because it makes what is really a very simple implementation look much more complex than it really is :)

Reply:
Yeah, that makes sense!